### PR TITLE
List repo on Artifact Hub; test add Velero policies

### PR DESCRIPTION
**Pull request by guohaoyu110**
## Related Issue(s)
https://github.com/artifacthub/hub/issues/2653#issuecomment-1377723000
#491 

## Description
Adding artifacthub-repo.yml and artifacthub-pkg.yml to enable users to easily discover, install and use the Velero policies on Artifact Hub. This will allow users to find the policies more easily and provide information about the policies such as the name, version, description, keywords, and creation date. Additionally, it will also enable users to get updates about the policies through the Artifact Hub interface.

## Checklist

- [x] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [x] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.

### CODE REVIEW

**Review by chipzoller — CHANGES_REQUESTED** [review]
1. `artifacthub-repo.yml` needs to go in the root of this repo.
2. In `artifacthub-repo.yml` remove all the comments, remove the `ignore[]` array, and add `owners[]` using my own contact info and name.
3. In each `artifacthub-pkg.yml` file, it needs more info that what is provided here. Annotations need to be created as per [here](https://artifacthub.io/docs/topics/annotations/kyverno/).

**Comment by chipzoller** [comment]
Make sure you've seen all my comments in #491

**Comment by guohaoyu110** [comment]
@chipzoller ok

**Comment by chipzoller** [comment]
Please also update the description. Your PRs should describe the changes that you are proposing by these changes. It should not be a copy-and-paste from a related project even if relevant.

**Comment by guohaoyu110** [comment]
> Please also update the description. Your PRs should describe the changes that you are proposing in only this PR. It should not be a copy-and-paste from a related project even if relevant.

thanks. changed.

**Comment by chipzoller** [comment]
Note that I also added the issue from this repo in the Related Issues field. PRs typically address open issues in the repo in which they're updated. Keep this in mind for the future. They may also reference issues from other repos, but in this case these PRs serve to address a pending item captured in an issue here.

**Comment by chipzoller** [comment]
Please also go through the schema of the `artifacthub-pks.yml` file [here](https://github.com/artifacthub/hub/blob/master/docs/metadata/artifacthub-pkg.yml) and try to complete the files to the best of your ability. You should also add the annotations as noted above.

**Comment by chipzoller** [comment]
Please let me know when you have addressed all my comments.

**Comment by guohaoyu110** [comment]
> Please let me know when you have addressed all my comments.

updated, please review.

File: artifacthub-repo.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+repositoryID: 231d81b2-c91a-4a4b-907e-67c0a863e8fa
+owners:
+  - name: Chip Zoller
+    email: chip@nirmata.com
```
Comment (chipzoller) [inline artifacthub-repo.yml:5]: chipzoller@gmail.com

File: velero/backup-all-volumes/artifacthub-pkg.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# changed if the `index.yaml` file changes. For git based repositories, it does
+# when the hash of the last commit in the branch you set up changes. This does
+# NOT apply to ownership claim operations, which are processed immediately.
+#
+---
+name: backup-all-volumes
+version: 1.0.0
+displayName: Backup All Volumes
+createdAt: 2023-03-29T00:00:00.000Z
+description: |
```
Comment (chipzoller) [inline velero/backup-all-volumes/artifacthub-pkg.yml:17]: The value of the description field should be the contents of the `policies.kyverno.io/description` annotation.

File: velero/backup-all-volumes/artifacthub-pkg.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  - backup
+  - volume
+  - kyverno
+license: Apache-2.0
+maintainers:
+  - name: Chip Zoller
+    email: chip@nirmata.com
+
+annotations: # (optional, keys and values must be strings)
+  kyverno/categories: "Velero"
```
Comment (chipzoller) [inline velero/backup-all-volumes/artifacthub-pkg.yml:34]: Key is incorrect according to documentation. It should be `kyverno/category`.

File: velero/backup-all-volumes/artifacthub-pkg.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  kyverno/kubernetesVersion: "1.20"
```
Comment (chipzoller) [inline velero/backup-all-volumes/artifacthub-pkg.yml:35]: This should be the value of the annotation `kyverno.io/kubernetes-version` in the policy itself, which is not correct as shown here.

File: velero/backup-all-volumes/artifacthub-pkg.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  kyverno/subject: "Volume"
```
Comment (chipzoller) [inline velero/backup-all-volumes/artifacthub-pkg.yml:36]: This should be the value of the annotation `policies.kyverno.io/subject` in the policy, which is not correct as shown here.

File: velero/backup-all-volumes/artifacthub-pkg.yml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  kyverno/version: "1.0.0"
```
Comment (chipzoller) [inline velero/backup-all-volumes/artifacthub-pkg.yml:37]: This should be the value, if present, of the annotation `policies.kyverno.io/minversion` in the policy, which is neither correct nor present as shown here.

File: velero/backup-all-volumes/artifacthub-pkg.yml [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  This policy backs up all volumes in a namespace.
+readme: |
+  This policy backs up all volumes in a namespace.
+  To use this policy, apply it to the namespace you want to backup.
+# Optional fields
+keywords:
+  - velero
+  - backup
+  - volume
+  - kyverno
```
Comment (chipzoller) [inline velero/backup-all-volumes/artifacthub-pkg.yml:27]: Let keywords be equal to `kyverno` and, in addition, whatever the value(s) of the annotation `policies.kyverno.io/category` as given in the policy. In this case, the keywords should equal

```yaml
- velero
- kyverno
```

File: velero/backup-all-volumes/artifacthub-pkg.yml [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+license: Apache-2.0
+maintainers:
+  - name: Chip Zoller
+    email: chip@nirmata.com
```
Comment (chipzoller) [inline velero/backup-all-volumes/artifacthub-pkg.yml:31]: Eliminate

**Review by chipzoller — CHANGES_REQUESTED** [review]
Please also add the `install` field as shown [here](https://github.com/artifacthub/hub/blob/master/docs/metadata/artifacthub-pkg.yml#L35-L38) and set the value as shown in an example [here](https://github.com/Ashwin901/policy-hub-automation/blob/master/policies/image-pullPolicy/artifacthub-pkg.yml#L11-L13).

**Comment by chipzoller** [comment]
Please address all comments and click "Resolve" for every one that is addressed in subsequent commits made.

File: velero/backup-all-volumes/artifacthub-pkg.yml [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+readme: |
+  This policy backs up all volumes in a namespace.
+  To use this policy, apply it to the namespace you want to backup.
+# Optional fields
+keywords:
+  - velero
+  - backup
+  - volume
+  - kyverno
+license: Apache-2.0
```
[the remark this answers is not in the document]
Comment (guohaoyu110) [reply velero/backup-all-volumes/artifacthub-pkg.yml]: resolved

File: velero/backup-all-volumes/artifacthub-pkg.yml (hunk 6 above)
Comment (guohaoyu110) [reply velero/backup-all-volumes/artifacthub-pkg.yml]: So, i should delete kyverno/version since it doesn't exist, right?

Comment (chipzoller) [reply velero/backup-all-volumes/artifacthub-pkg.yml]: Yes

File: velero/backup-all-volumes/artifacthub-pkg.yml [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+createdAt: "2023-03-29T00:00:00.000Z"
+description: >-
+      In order for Velero to backup volumes in a Pod using an opt-in approach, it
+      requires an annotation on the Pod called `backup.velero.io/backup-volumes` with the
+      value being a comma-separated list of the volumes mounted to that Pod. This policy
+      automatically annotates Pods (and Pod controllers) which refer to a PVC so that
+      all volumes are listed in the aforementioned annotation if a Namespace with the label
+      `velero-backup-pvc=true`.
+install: |-
+  ### Usage
```
Comment (chipzoller) [inline velero/backup-all-volumes/artifacthub-pkg.yml:14]: Remove all comments.

**Review by chipzoller — CHANGES_REQUESTED** [review]
See last round of comments. Be sure to apply those comments to all metadata files. Also, when making commits, the commit message should be unique and descriptive for the changes within. I've noticed you are using the same messages over again which should not be done.

**Comment by guohaoyu110** [comment]
@chipzoller updated, thanks.

**Review by chipzoller — APPROVED** [review]

**Comment by chipzoller** [comment]
@guohaoyu110, now these are listed, take a look at one [here](https://artifacthub.io/packages/kyverno/kyverno-policies/backup-all-volumes). We should add the `readme` field in `artifacthub-pkg.yml` as shown [here](https://github.com/artifacthub/hub/blob/master/docs/metadata/artifacthub-pkg.yml#L31-L34). Please add this for all 3 policies reflected in this PR and set their values to the same as the description followed by a new line followed by this sentence:

```
Refer to the documentation for more details on Kyverno annotations: https://artifacthub.io/docs/topics/annotations/kyverno/
```

Please submit these changes in a new PR.

**Comment by guohaoyu110** [comment]
@chipzoller  ok

**Comment by guohaoyu110** [comment]
> @guohaoyu110, now these are listed, take a look at one [here](https://artifacthub.io/packages/kyverno/kyverno-policies/backup-all-volumes). We should add the `readme` field in `artifacthub-pkg.yml` as shown [here](https://github.com/artifacthub/hub/blob/master/docs/metadata/artifacthub-pkg.yml#L31-L34). Please add this for all 3 policies reflected in this PR and set their values to the same as the description followed by a new line followed by this sentence:
> 
> ```
> Refer to the documentation for more details on Kyverno annotations: https://artifacthub.io/docs/topics/annotations/kyverno/
> ```
> 
> Please submit these changes in a new PR.

submitted.

**Pull request merged**